extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/metrics/MongoMetricsConnectionPoolListener.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.mongodb.metrics;
+
+import org.eclipse.microprofile.metrics.Counter;
+import org.eclipse.microprofile.metrics.Metadata;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.eclipse.microprofile.metrics.MetricType;
+import org.eclipse.microprofile.metrics.Tag;
+
+import com.mongodb.connection.ServerId;
+import com.mongodb.event.ConnectionAddedEvent;
+import com.mongodb.event.ConnectionCheckedInEvent;
+import com.mongodb.event.ConnectionCheckedOutEvent;
+import com.mongodb.event.ConnectionPoolClosedEvent;
+import com.mongodb.event.ConnectionPoolListener;
+import com.mongodb.event.ConnectionPoolOpenedEvent;
+import com.mongodb.event.ConnectionPoolWaitQueueEnteredEvent;
+import com.mongodb.event.ConnectionPoolWaitQueueExitedEvent;
+import com.mongodb.event.ConnectionRemovedEvent;
+
+import io.smallrye.metrics.MetricRegistries;
+
+public class MongoMetricsConnectionPoolListener implements ConnectionPoolListener {
+    private final static String sizeName = "mongodb.connection-pool.size";
+    private final static String checkedOutCountName = "mongodb.connection-pool.checked-out-count";
+
+    @Override
+    public void connectionPoolOpened(ConnectionPoolOpenedEvent event) {
+        Tag[] tags = createTags(event.getServerId());
+
+        Metadata sizeMetadata = Metadata.builder().withName(sizeName).withType(MetricType.COUNTER)
+                .withDescription("the current size of the pool, including idle and and in-use members").build();
+        Metadata checkedOutCountMetadata = Metadata.builder().withName(checkedOutCountName).withType(MetricType.COUNTER)
+                .withDescription(" the current count of connections that are currently in use").build();
+
+        getMetricRegistry().register(sizeMetadata, new ConnectionPoolCounter(), tags);
```

Comment:
In development mode, after a reload, the metrics from previous run may still exist in the registry. At the moment we don't automatically deregister vendor metrics during reload (I think we should, I will take care of that soon, but for now we don't). So temporarily this should probably contain a check if the metric already exists, and if yes, it probably means that we just reloaded and the metric should be removed and re-registered (so it will start again from zero).